packages/synapse-sdk/src/storage/context.ts

```diff
@@ -835,26 +836,29 @@ export class StorageContext {
   /**
    * Upload data to the service provider
    */
-  async upload(data: Uint8Array | ArrayBuffer, options?: UploadOptions): Promise<UploadResult> {
+  async upload(data: Uint8Array | ArrayBuffer, options?: UploadOptions, pieceCid?: PieceCID): Promise<UploadResult> {
```

Comment:
I've just realised that I should have called out this breaking API, `pieceCid` should be in `UploadOptions`

In my streaming PR I'm putting it in there, so we'll have a break & unbreak